src/util/events.rs

```diff
@@ -100,6 +100,16 @@ pub enum Event {
 	BroadcastChannelUpdate {
 		msg: msgs::ChannelUpdate,
 	},
+	/// Used to tell a peer that something is incorrect
+	SendErrorMessage {
+		node_id: PublicKey,
+		msg: msgs::ErrorMessage,
+	},
+	// Events indicating the network loop should change the state of peer connection:
+	/// Used to indicate that the peer connection should be closed with a message holding the reason
+	DisconnectPeer {
```

Comment:
Why not just make it a bool in SendErrorMessage as to whether to disconnect? Feels like thats usually what you'd want, no?